internal/vercheck/vercheck.go

```diff
@@ -50,10 +61,8 @@ func SelfUpdate(stdOut, stdErr io.Writer) error {
 		return usererr.New("curl or wget is required to update devbox. Please install either and try again.")
 	}
 
-	// Delete version cache. Keep this in-sync with whatever logic is in launch.sh
-	cacheDir := xdg.CacheSubpath("devbox")
-	versionCacheFile := filepath.Join(cacheDir, "latest-version")
-	_ = os.Remove(versionCacheFile)
+	// Delete version cache.
+	_ = os.Remove(versionCacheFilePath())
```

Comment:
I think we should handle this error. If this file can't be deleted we may end up in a bad state:
```golang
if err := os.Remove(versionCacheFilePath()); err !=nil && !os.IsNotExist(err) {
  return ...
}
```